apps/web/components/eventtype/AIEventController.tsx

```diff
@@ -85,10 +89,64 @@ export default function AIEventController({ eventType, isTeamEvent }: AIEventCon
   );
 }
 
+const ErrorMessage = ({ fieldName, message }: { fieldName: string; message: string }) => {
+  const { t } = useLocale();
+  return (
+    <div data-testid={`error-message-${fieldName}`} className="mt-2 flex items-center text-sm text-red-700 ">
+      <Icon name="info" className="h-3 w-3 ltr:mr-2 rtl:ml-2" />
+      <p>{t(message || "invalid_input")}</p>
+    </div>
+  );
+};
+
+const TemplateFields = () => {
+  const { t } = useLocale();
+  const formMethods = useFormContext<FormValues>();
+  const { control, watch } = formMethods;
+
+  const templateType = watch("aiPhoneCallConfig.templateType");
+  const fields = TEMPLATES_FIELDS[templateType as TemplateType];
+
+  return (
+    <div className="space-y-4">
+      {fields?.map((field) => (
+        <div key={field.name}>
+          <Controller
+            control={control}
+            name={`aiPhoneCallConfig.${field.name}`}
+            render={({ field: { value, onChange }, fieldState: { error } }) => {
+              const { variableName, ...restField } = field;
+              const variableInfo = !!variableName ? `: ${t("variable")} {{${variableName}}}` : "";
+              return (
+                <div>
+                  <ComponentForField
+                    field={{
+                      ...restField,
+                      label: `${t(field.defaultLabel)}${variableInfo}`,
+                      placeholder: t(field.placeholder),
+                    }}
+                    value={value ?? ""}
+                    readOnly={false}
+                    setValue={(val: unknown) => {
+                      onChange(val);
+                    }}
+                  />
+                  {error?.message && <ErrorMessage message={error.message} fieldName={field.name} />}
+                </div>
+              );
+            }}
+          />
+        </div>
+      ))}
+    </div>
+  );
+};
+
 const AISettings = ({ eventType }: { eventType: EventTypeSetup }) => {
   const { t } = useLocale();
 
   const formMethods = useFormContext<FormValues>();
+
   const [calApiKey, setCalApiKey] = useState("");
```

Comment:
> Should we also have a drop of available API keys under the user? It would save them time from navigating to different pages.

I am still trying to figure out how can we implement this as API key is encrypted and then stored in DB.